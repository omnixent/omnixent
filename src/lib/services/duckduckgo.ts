import { ServiceArgs } from '.';
import { normalizeServiceArgs } from '../utils';

const endpoint = 'https://duckduckgo.com/ac/';

export function formatURI(options: ServiceArgs): string {
  const { t, hl } = normalizeServiceArgs(options);
  return `${endpoint}?client=psy-ab&q=${t}&kl=${hl}`;
}

export function extractBody(response: string) {
  return JSON.parse(response)?.map(({ phrase }: any) => phrase);
}
