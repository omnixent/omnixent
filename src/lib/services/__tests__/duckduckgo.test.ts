import { formatURI, extractBody } from '../duckduckgo';

test('Testing google formatURI', () => {
  expect(formatURI({ term: 'Node.js', language: 'it', country: 'us' })).toMatchInlineSnapshot(
    `"https://duckduckgo.com/ac/?client=psy-ab&q=node.js&kl=us-IT"`,
  );
  expect(
    formatURI({ term: 'blue jeans brands', language: 'en', country: 'de' }),
  ).toMatchInlineSnapshot(
    `"https://duckduckgo.com/ac/?client=psy-ab&q=blue%20jeans%20brands&kl=de-EN"`,
  );
});

test('Testing google extractBody', () => {
  expect(
    extractBody(
      `[{"phrase":"node.js"},{"phrase":"node.js download"},{"phrase":"node.js mongodb"},{"phrase":"node.js tutorial"},{"phrase":"node.js fs"},{"phrase":"node.js mysql"},{"phrase":"node.js ubuntu"},{"phrase":"node.js project"}]`,
    ),
  ).toMatchSnapshot();
  expect(
    extractBody(
      `[{"phrase":"blue jeans brands"},{"phrase":"blue jeans brands list"},{"phrase":"blue jeans brand names"}]`,
    ),
  ).toMatchSnapshot();
});
