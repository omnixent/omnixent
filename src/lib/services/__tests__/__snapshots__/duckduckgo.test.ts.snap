// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing google extractBody 1`] = `
Array [
  "node.js",
  "node.js download",
  "node.js mongodb",
  "node.js tutorial",
  "node.js fs",
  "node.js mysql",
  "node.js ubuntu",
  "node.js project",
]
`;

exports[`Testing google extractBody 2`] = `
Array [
  "blue jeans brands",
  "blue jeans brands list",
  "blue jeans brand names",
]
`;
